### PR TITLE
Monitoring example for Routing Service added.

### PR DESCRIPTION
**Pull request by IreneAlejoTeissiere**
### Summary

I added an example of how to subscribe to the monitoring topics for Routing Service

### Details and comments

I may need some guidance regarding the code style.
Do I need to sign the CONTRIBUTING form?

### Checks

-   [x] I have updated the documentation accordingly.
-   [x] I have read the [CONTRIBUTING](https://github.com/rticommunity/rticonnextdds-examples/blob/develop/CONTRIBUTING.md) document.

### CODE REVIEW

File: examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+
+#include "RoutingServiceMonitoring.hpp"
+
+// Get the Configuration sample (rti/service/monitoring/config topic) from its
+// Instance Handle. The reader for that topic is passed as a parameter. Also the
+// Instance Handle.
+RTI::RoutingService::Monitoring::Config getConfig(
+        dds::sub::DataReader<RTI::RoutingService::Monitoring::Config>
+                &configReader,
+        const dds::core::InstanceHandle &handle)
```
Comment (Antoniolm) [inline examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx:33]: Reference should be located beside the type
```suggestion
        const dds::core::InstanceHandle& handle)
```
Same for line 31-32 parameter. I could see a few more reference issues below.

File: examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+{
+    // Get the configuration for an instance handle to use this information to
+    // complement what is received in the event or periodic topic
+    auto cSamples = configReader.select()
+                            .instance(handle)
+                            .state(dds::sub::status::DataState::any_data())
+                            .read();
```
Comment (Antoniolm) [inline examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx:40]: Only two levels of indentation
```suggestion
            .instance(handle)
            .state(dds::sub::status::DataState::any_data())
            .read();
```
Same issue below in a few more lines.

File: examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+    if (cSamples.length() == 0) {
+        std::stringstream errorMsg;
+        errorMsg << "The configuration for the instance handle " << handle
+                 << " has not being received";
+        throw std::runtime_error((errorMsg.str()));
+    }
+    return cSamples[cSamples.length() - 1].data();  // We are interested only in
```
Comment (Antoniolm) [inline examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx:47]: Trailing comments can only have one single line. We should put this comment above the return statement.

File: examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+                    auto config = getConfig(
+                            configReader,
+                            sample.info().instance_handle());
+                    std::cout << "Periodic data:\n\tDomain Route: ";
+                    if (config.value()
+                                .routing_domain_route()
+                                .connections()
+                                .value()
+                                .size()
+                        > 0) {
```
Comment (Antoniolm) [inline examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx:81]: To increase readability, we can move the `> 0) {` like this:
```suggestion
                            .size() > 0) {
```

File: examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+                    qosProvider.datareader_qos(
+                            "monitoring_Library::periodic_Profile"));
+    // Create a ReadCondition for any data on this reader and associate a
+    // handler
+    int count = 0;
+    dds::sub::cond::ReadCondition eventReadCondition(
+            eventReader,
+            dds::sub::status::DataState::any(),
+            [&eventReader, &configReader, &count](
+                    /* dds::core::cond::Condition condition */) {
```
Comment (Antoniolm) [inline examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx:329]: Is this comment required? Same at line 336

File: examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+{
+    int count = 0;
+    try {
+        // Take all samples
+        dds::sub::LoanedSamples<RTI::RoutingService::Monitoring::Event>
+                samples = eventReader.take();
+        for (const auto &sample : samples) {
+            if (sample.info().valid()) {
+                count++;
+                if (sample.data().value()._d()
```
Comment (Antoniolm) [inline examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx:164]: I had been thinking about this if-else condition around the sample data and I think that we can use a switch-case scenario to improve readability.
What do you think? I didn't go deep in the code and I probably miss something. Please let me know your thoughts :)

File: examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+    } catch (const dds::core::Exception &e) {
+        std::cerr << "A Connext exception has been thrown: " << e.what()
+                  << std::endl;
+    } catch (const std::exception &e) {
+        std::cerr << "An std exception has been thrown: " << e.what()
+                  << std::endl;
+    }
+
+    return count;
+}  // The LoanedSamples destructor returns the loan
```
Comment (Antoniolm) [inline examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx:285]: We can move this comment before the return

File: examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+{
+    int count = 0;
+    try {
+        // Take all samples
+        dds::sub::LoanedSamples<RTI::RoutingService::Monitoring::Periodic>
+                samples = periodicReader.take();
+        for (const auto &sample : samples) {
+            if (sample.info().valid()) {
+                count++;
+                if (sample.data().value()._d()
```
Comment (Antoniolm) [inline examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx:69]: The same case that below about the switch-case for readability.

**Review by Antoniolm — CHANGES_REQUESTED** [review]

File: examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx (hunk 6 above)
Comment (IreneAlejoTeissiere) [reply examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx:164]: Thanks for the feedback! I also think that we can use a switch here. In these cases, I usually prefer if/else but that is only a personal preference. Let's say that I am not such a big fan of switch statements, but it is not them, it's me ;)

I will change it!

File: examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx (hunk 5 above)
Comment (IreneAlejoTeissiere) [reply examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx:329]: It comes with the autogenerated code I used as a base. I think we can remove it. It doesn't make sense in this example

File: examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx (hunk 2 above)
Comment (IreneAlejoTeissiere) [reply examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx:40]: I am wondering if I am using the same template for indenting the code as you. I am using VSCode and it should be taking the one that is in the repository. Let's talk about it via Slack.

**Review by Antoniolm — DISMISSED** [review]

**Comment by ManuelJNunez** [comment]
Hi Irene, the README you added is not following the Markdown template for `C/C++` examples. You can see an example [here](https://github.com/rticommunity/rticonnextdds-examples/tree/master/examples/connext_dds/asynchronous_publication/c%2B%2B03).

File: examples/routing_service/routing_service_monitoring/c++11/README.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+## Running the Example
+
+In two separate command prompt windows for the publisher and subscriber. Run the
+following commands from the example directory (this is necessary to ensure the
+application loads the QoS defined in *USER_QOS_PROFILES.xml*):
+
+On *Windows* systems run:
+
+```sh
+RoutingServiceMonitoring_subscriber.exe <domain_id>
```
Comment (ManuelJNunez) [inline examples/routing_service/routing_service_monitoring/c++11/README.md:45]:
```sh
RoutingServiceMonitoring_subscriber.exe <domain_id> <sample_count>
```

File: examples/routing_service/routing_service_monitoring/c++11/README.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+```
+
+On *UNIX* systems run:
+
+```sh
+./RoutingServiceMonitoring_subscriber <domain_id>
```
Comment (ManuelJNunez) [inline examples/routing_service/routing_service_monitoring/c++11/README.md:51]:
```sh
./RoutingServiceMonitoring_subscriber <domain_id> <sample_count>
```

**Review by ManuelJNunez — CHANGES_REQUESTED** [review]

File: examples/routing_service/routing_service_monitoring/c++11/README.md (hunk 9 above)
Comment (IreneAlejoTeissiere) [reply examples/routing_service/routing_service_monitoring/c++11/README.md]: ok!

File: examples/routing_service/routing_service_monitoring/c++11/README.md (hunk 10 above)
Comment (IreneAlejoTeissiere) [reply examples/routing_service/routing_service_monitoring/c++11/README.md]: Ok!

**Review by lulivi — DISMISSED** [review]

**Review by ManuelJNunez — APPROVED** [review]

File: examples/routing_service/routing_service_monitoring/c++11/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,93 @@
+# (c) 2019 Copyright, Real-Time Innovations, Inc. All rights reserved.
```
Comment (fgaranda) [inline examples/routing_service/routing_service_monitoring/c++11/CMakeLists.txt:1]: Replace copyright notice with:
```
# (c) 2021 Copyright, Real-Time Innovations, Inc.  All rights reserved.
# RTI grants Licensee a license to use, modify, compile, and create derivative
# works of the software solely for use with RTI Connext DDS.  Licensee may
# redistribute copies of the software provided that all such copies are
# subject to this license. The software is provided "as is", with no warranty
# of any type, including any warranty for fitness for any purpose. RTI is
# under no obligation to maintain or support the software.  RTI shall not be
# liable for any incidental or consequential damages arising out of the use or
# inability to use the software.
```

File: examples/routing_service/routing_service_monitoring/c++11/USER_QOS_PROFILES.xml [hunk 12]
```diff
@@ -0,0 +1,295 @@
...
+RTI grants Licensee a license to use, modify, compile, and create derivative
+works of the software solely for use with RTI Connext DDS. Licensee may
+redistribute copies of the software provided that all such copies are
+subject to this license. The software is provided "as is", with no warranty
+of any type, including any warranty for fitness for any purpose. RTI is
+under no obligation to maintain or support the software. RTI shall not be
+liable for any incidental or consequential damages arising out of the use
+or inability to use the software.
+
+The QoS configuration of the DDS entities in the generated example is loaded
```
Comment (fgaranda) [inline examples/routing_service/routing_service_monitoring/c++11/USER_QOS_PROFILES.xml:16]: You can get rid of all the boiler plate comments and leave just the usual:
```xml
<!-- 
 (c) 2021 Copyright, Real-Time Innovations, Inc.  All rights reserved.
 RTI grants Licensee a license to use, modify, compile, and create derivative
 works of the Software.  Licensee has the right to distribute object form only
 for use with RTI products.  The Software is provided "as is", with no warranty
 of any type, including any warranty for fitness for any purpose. RTI is under
 no obligation to maintain or support the Software.  RTI shall not be liable for
 any incidental or consequential damages arising out of the use or inability to
 use the software.
```

File: examples/routing_service/routing_service_monitoring/c++11/USER_QOS_PROFILES.xml [hunk 13]
```diff
@@ -0,0 +1,295 @@
...
+profile you wish to modify. The NDDS_QOS_PROFILES.example.xml file (contained in 
+the same directory as the BuiltinProfiles.documentationONLY.xml file) shows how
+to inherit from the built-in profiles. 
+
+For more information about XML QoS Profiles see the "Configuring QoS with
+XML" chapter in the RTI Connext DDS Core Libraries User's Manual.
+-->
+
+<dds xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+     xsi:noNamespaceSchemaLocation="/Applications/rti_connext_dds-6.0.1/resource/schema/rti_dds_qos_profiles.xsd">
```
Comment (fgaranda) [inline examples/routing_service/routing_service_monitoring/c++11/USER_QOS_PROFILES.xml:125]: Replace with the following URL: `http://community.rti.com/schema/6.1.0/rti_dds_qos_profiles.xsd`

**Review by fgaranda — CHANGES_REQUESTED** [review]

File: examples/routing_service/routing_service_monitoring/RSConfig.xml [hunk 14]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0"?>
```
[the remark this answers is not in the document]
Comment (IreneAlejoTeissiere) [reply examples/routing_service/routing_service_monitoring/RSConfig.xml:1]: Done!

File: examples/routing_service/routing_service_monitoring/c++11/CMakeLists.txt (hunk 11 above)
Comment (IreneAlejoTeissiere) [reply examples/routing_service/routing_service_monitoring/c++11/CMakeLists.txt]: Done!

File: examples/routing_service/routing_service_monitoring/c++11/USER_QOS_PROFILES.xml (hunk 12 above)
Comment (IreneAlejoTeissiere) [reply examples/routing_service/routing_service_monitoring/c++11/USER_QOS_PROFILES.xml]: Done!

File: examples/routing_service/routing_service_monitoring/c++11/USER_QOS_PROFILES.xml (hunk 13 above)
Comment (IreneAlejoTeissiere) [reply examples/routing_service/routing_service_monitoring/c++11/USER_QOS_PROFILES.xml]: Done!

**Review by lulivi — APPROVED** [review]

**Review by fgaranda — APPROVED** [review]

**Pull request merged**